text/preprocessing-api-rework.md

```diff
@@ -55,8 +66,9 @@ function extractStyles(code: string): Array<{
     end: number;
   }>;
 }>;
+```
 
-extracts the style tags from the source code, each with start/end position, content and attributes
+extracts the style tags from the source code, each with start/end position, content and attributes. `isTopLevel` would require a parser which has to make some assumptions about the code being "standard JS/HTML-syntax compliant" (in the sense of opening closing brackets match etc). Pending PR for such a parser here: https://github.com/sveltejs/svelte/pull/6611 . We could make using that parser the default, with a fallback to the old regex-approach in case of an error, in which case `isTopLevel` would be `undefined`, not `true` or `false`.
```

Comment:
Three-state-boolean warning. Perhaps we could do this instead:

```ts
locationStatus: "top" | "nested" | "unknown"
```

That name (`locationStatus`) is maybe not the best since I am not familiar with this problem directly. In any case, I'd personally avoid any type where `true`, `false` and `undefined` are three separate meanings.